### ISSUE TITLE
Exclude certain matches from search results

### ISSUE THREAD

**Issue by AucaCoyan** [federico-terzi/espanso#2134]
@melissis have a point here. Maybe it will be implemented, maybe not, but we first need to discuss if it's wanted 😊

### Discussed in https://github.com/espanso/espanso/discussions/1494

<div type='discussions-op-text'>

<sup>Originally posted by **melissis** December 31, 2022</sup>
## Suggestion
I find it annoying that rules from my misspell configs show up in the search box. I therefore request there be a way to exclude certain matches from search results, so that they will never appear in the search box.
Preferably, there should both be a config-wide configuration parameter (to exclude everything in a yml), and a match property (to exclude single rules). I would prioritize the former, due to its ease of use, but the latter would, in a way, be more powerful.

### Example
I want to find emojis with triggers containing the word "fire".

![2023-01-01 02_06_38-](https://user-images.githubusercontent.com/116532973/210158593-8cdb2a55-debf-47fc-869f-88c6b05970ce.png)

I have to scroll all the way to the bottom.

EDIT: I guess I could search with ":" and it would skip these entries... But maybe I'm not the only one who went a long time without realizing this. And it _does_ take a 0.x seconds longer to write that out. I still think my suggestion is solid.</div>

**Comment by smeech** [comment]
It looks like a lot of work for something that's easily worked around.

**Comment by AucaCoyan** [comment]
Yeah, I thought so. I added the `low priority` label

**Comment by Waslain** [comment]
I agree about this it would be really nice to be able to add a filter at least to a whole package. In the code it will just be bool check before appending all the triggers in the menu list. This way we could exclude misspell, translation packages that are a real mess from the standard list.
